src/adapters/ReactFifteenAdapter.js

```diff
@@ -0,0 +1,205 @@
+}
+
+class SimpleWrapper extends React.Component {
+  render() {
+    return this.props.node || null;
+  }
+}
+
+SimpleWrapper.propTypes = { node: PropTypes.node.isRequired };
+
+class ReactFifteenAdapter extends EnzymeAdapter {
+  createMountRenderer(options) {
+    assertDomAvailable('mount');
+    const domNode = options.attachTo || global.document.createElement('div');
+    let instance = null;
+    return {
+      render(el/* , context */) {
+        const wrappedEl = React.createElement(SimpleWrapper, {
+          node: el,
+        });
+        instance = ReactDOM.render(wrappedEl, domNode);
+      },
+      unmount() {
+        ReactDOM.unmountComponentAtNode(domNode);
+      },
+      getNode() {
+        return instanceToTree(instance._reactInternalInstance._renderedComponent);
+      },
+      simulateEvent(node, event, mock) {
+        const mappedEvent = mapNativeEventNames(event);
+        const eventFn = TestUtils.Simulate[mappedEvent];
+        if (!eventFn) {
+          throw new TypeError(`ReactWrapper::simulate() event '${event}' does not exist`);
+        }
+        // eslint-disable-next-line react/no-find-dom-node
+        eventFn(ReactDOM.findDOMNode(node.instance), mock);
+      },
+      batchedUpdates(fn) {
+        return ReactDOM.unstable_batchedUpdates(fn);
+      },
+    };
+  }
+
+  createShallowRenderer(/* options */) {
+    const renderer = TestUtils.createRenderer();
+    let isDOM = false;
+    let cachedNode = null;
+    return {
+      render(el, context) {
+        cachedNode = el;
+        /* eslint consistent-return: 0 */
+        if (typeof el.type === 'string') {
+          isDOM = true;
+        } else {
+          isDOM = false;
+          return withSetStateAllowed(() => renderer.render(el, context));
+        }
+      },
+      unmount() {
+        renderer.unmount();
+      },
+      getNode() {
+        if (isDOM) {
+          return elementToTree(cachedNode);
+        }
+        const output = renderer.getRenderOutput();
+        return {
+          nodeType: 'class',
+          type: cachedNode.type,
+          props: cachedNode.props,
+          key: cachedNode.key,
+          ref: cachedNode.ref,
+          instance: renderer._instance._instance,
+          rendered: elementToTree(output),
+        };
+      },
+      simulateEvent(node, event, ...args) {
+        const handler = node.props[propFromEvent(event)];
+        if (handler) {
+          withSetStateAllowed(() => {
+            // TODO(lmr): create/use synthetic events
+            // TODO(lmr): emulate React's event propagation
+            renderer.unstable_batchedUpdates(() => {
+              handler(...args);
+            });
+          });
+        }
+      },
+      batchedUpdates(fn) {
+        return withSetStateAllowed(() => renderer.unstable_batchedUpdates(fn));
+      },
+    };
+  }
+
+  createStringRenderer(/* options */) {
+    return {
+      render(el /* , context */) {
```

Comment:
It doesn't actually matter here, since the context is lifted in `render.jsx`. `context` will never have a value here with the way it is currently written. This could be refactored later if we wanted, but it wouldn't actually affect the public API at all.